polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -2196,6 +2198,60 @@ impl State {
 
 		sanitized_assignments
 	}
+
+	// Filter out obviously invalid candidate indicies.
+	async fn sanitize_v1_approvals(
+		&mut self,
+		peer_id: PeerId,
+		sender: &mut impl overseer::ApprovalDistributionSenderTrait,
+		approval: Vec<IndirectSignedApprovalVote>,
+	) -> Vec<IndirectSignedApprovalVoteV2> {
+		let mut sanitized_approvals = Vec::new();
```

Comment:
to avoid additional allocation, you can do this in-place, by iterating in reverse order (.into_iter().rev()) and `.swap_remove` to remove invalid votes while maintaining linear complexity. it might change the order of votes, but we don't care about that AFAIK